vllm/inputs/preprocess.py

```diff
@@ -298,68 +330,33 @@ def _prompt_to_llm_inputs(
         * :class:`SingletonInputs` instance
         """
         parsed = parse_singleton_prompt(prompt)
-
-        if parsed["type"] == "str":
-            prompt_text = parsed["content"]
-            prompt_token_ids = self._tokenize_prompt(
-                prompt_text,
+        prompt_text, prompt_token_ids, token_type_ids, cache_salt = \
+            self._get_prompt_data(parsed)
+
+        # If multimodal data is present, process and return immediately
+        if parsed["type"] != "str" and "multi_modal_data" in parsed[
+                "content"] is not None:
```

Comment:
```suggestion
        if parsed["type"] != "str" and parsed["content"].get("multi_modal_data"] is not None:
```
The failure is because of this

Reply:
Oh, thanks!